packages/arquero-react/src/ArqueroDetailsList/__tests__/RowGroupingTestStory/RowGroupingTestStory.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ColumnMetadata, TableMetadata } from '@essex/arquero'
+import { introspect } from '@essex/arquero'
+import { ArqueroDetailsList, ArqueroTableHeader } from '@essex/arquero-react'
+import type { IDetailsGroupDividerProps } from '@fluentui/react'
+import { Label, PrimaryButton } from '@fluentui/react'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+import { memo, useCallback, useEffect, useState } from 'react'
+
+import { createLazyLoadingGroupHeader } from '../component-factories.js'
+import { ButtonContainer, Table } from './RowGroupingTestStory.styles.js'
+
+export interface RowGroupingTestStoryProps {
+	mockTable: ColumnTable | undefined
+}
+
+export const RowGroupingTestStory: React.FC<RowGroupingTestStoryProps> = memo(
+	function RowGroupingTestStory({ mockTable }) {
+		const [groupedTable, setGroupedTable] = useState<ColumnTable | undefined>()
+		const [groupedMetadata, setGroupedMetadata] = useState<
+			TableMetadata | undefined
+		>()
+
+		const [groupBy, setGroupBy] = useState<string>('')
```

Comment:
This would be cleaner if the type was `string | undefined` and the default left undefined

Reply:
It would actually be better if the groupby was an array, so we could have _both_ columns as grouping and test that the hierarchical row nesting works properly. So clicking each button would check if their column was already in the list, and add it if not or remove if it is (i.e., acting as a "toggle")